magda-authorization-api/src/createApiRouter.ts

```diff
@@ -119,6 +127,56 @@ export default function createApiRouter(options: ApiRouterOptions) {
         next();
     };
 
+    /**
+     * retrieve user info with api key id & api key
+     * this api is only meant to be accessed internally (by gateway)
+     * This route needs to run without MUST_BE_ADMIN middleware as it will authenticate request by APIkey itself
+     */
+    router.get("/private/getUserByApiKey/:apiKeyId", async function (req, res) {
```

Comment:
Generally we shouldn't have to put `get` in the path of a `GET` request. Could this be something a bit more conventionally RESTful like `/private/users/apikey/:apiKeyId`?

Reply:
Updated.